R/run_stress_test.R

```diff
@@ -1,55 +1,126 @@
 #' Run stress testing for provided asset type.
 #'
-#' @param asset_type String holding asset_type, for allowed value compare
+#' This function runs the transition risk stress test. It can be desirable to
+#' understand sensitivities of the scenarios, in which case the user may pass a
+#' vector of values to one (and only one) of the detail arguments. This will
+#' result in running the analysis multiple times in a row with the argument
+#' varied.
+#' NOTE: argument `asset_type_arg` is not iterateable.
+#'
+#' @param asset_type_arg String holding asset_type, for allowed value compare
 #'   `asset_types_lookup`.
-#' @param lgd_senior_claims Numeric, holding the loss given default for senior
+#' @param lgd_senior_claims_arg Numeric, holding the loss given default for senior
 #'   claims, for accepted value range check `lgd_senior_claims_range_lookup`.
-#' @param lgd_subordinated_claims Numeric, holding the loss given default for
+#' @param lgd_subordinated_claims_arg Numeric, holding the loss given default for
 #'   subordinated claims, for accepted value range check
 #'   `lgd_subordinated_claims_range_lookup`.
-#' @param terminal_value Numeric. A ratio to determine the share of the
+#' @param terminal_value_arg Numeric. A ratio to determine the share of the
 #'   discounted value used in the terminal value calculation beyond the
 #'   projected time frame. For accepted range compare `terminal_value_range_lookup`.
-#' @param risk_free_rate Numeric that indicates the risk free rate of interest.
+#' @param risk_free_rate_arg Numeric that indicates the risk free rate of interest.
 #'   For accepted range compare `risk_free_rate_range_lookup`.
-#' @param discount_rate Numeric, that holds the discount rate of dividends per
+#' @param discount_rate_arg Numeric, that holds the discount rate of dividends per
 #'   year in the DCF. For accepted range compare `discount_rate_range_lookup`.
-#' @param div_netprofit_prop_coef Numeric. A coefficient that determines how
+#' @param div_netprofit_prop_coef_arg Numeric. A coefficient that determines how
 #'   strongly the future dividends propagate to the company value. For accepted
 #'   range compare `div_netprofit_prop_coef_range_lookup`.
-#' @param shock_year Numeric, holding year the shock is applied. For accepted
+#' @param shock_year_arg Numeric, holding year the shock is applied. For accepted
 #'   range compare `shock_year_range_lookup`.
-#' @param term Numeric. A coefficient that determines for which maturity the
+#' @param term_arg Numeric. A coefficient that determines for which maturity the
 #'   expected loss should be calculated in the credit risk section. For accepted
 #'   range compare `term_range_lookup`.
-#' @param company_exclusion Boolean, indicating if companies provided in dataset
+#' @param company_exclusion_arg Boolean, indicating if companies provided in dataset
```

Comment:
Nit: Add "[the] dataset". In general write technical docs as you would normally write any prose.